### PR TITLE
feat: k8s clusters data source

### PR DESCRIPTION
**Pull request by cbeti-ionos**
## What does this fix or implement?
Data source for retrieving a list of k8s clusters which match filtering criteria.

## Checklist

- [x] PR name added as appropriate (e.g. `feat:`/`fix:`/`doc:`/`test:`/`refactor:`)
- [x] Tests added or updated
- [x] Documentation updated
- [x] Changelog updated and version incremented (label: upcoming release)
- [ ] Github Issue linked if any
- [ ] Jira task updated

### CODE REVIEW

File: docs/data-sources/k8_clusters.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+---
+subcategory: "Compute Engine"
+layout: "ionoscloud"
+page_title: "IonosCloud : k8s_clusters"
```
Comment (cristiGuranIonos) [inline docs/data-sources/k8_clusters.md:4]:
```suggestion
page_title: "IonosCloud : ionoscloud_k8s_clusters"
```

File: ionoscloud/data_source_k8s_clusters.go [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+	clusterConfigProperties["users"] = users
+
+	clusterProperties["config"] = []map[string]any{clusterConfigProperties}
+	clusterProperties["ca_crt"] = decodedCert
+	clusterProperties["user_tokens"] = userTokens
+
+	return clusterProperties, nil
+}
+
+func mergeMaps(maps ...map[string]any) map[string]any {
```
Comment (cristiGuranIonos) [inline ionoscloud/data_source_k8s_clusters.go:238]: should his be in utils? Doesn't seem to be k8s related

File: docs/data-sources/k8_clusters.md (hunk 1 above)
Comment (cbeti-ionos) [reply docs/data-sources/k8_clusters.md]: modified ([b9a7d92](https://github.com/ionos-cloud/terraform-provider-ionoscloud/pull/519/commits/b9a7d92cfbcc64f79aa318a0eeab6fc491584516))

File: ionoscloud/data_source_k8s_clusters.go (hunk 2 above)
Comment (cbeti-ionos) [reply ionoscloud/data_source_k8s_clusters.go]: agreed, moved to utils and added some comments ([b9a7d92](https://github.com/ionos-cloud/terraform-provider-ionoscloud/pull/519/commits/b9a7d92cfbcc64f79aa318a0eeab6fc491584516))

File: docs/data-sources/k8_clusters.md (hunk 1 above)
Comment (cristiGuranIonos) [reply docs/data-sources/k8_clusters.md]: you can just commit suggestion next time, should be easier.

**Comment by cristiGuranIonos** [comment]
Let's change cyclo to 25.

File: ionoscloud/data_source_k8s_clusters.go [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+		req.Filter(key, value)
+	}
+
+	clusters, apiResponse, err := req.Execute()
+	logApiRequestTime(apiResponse)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("an error occurred while fetching k8s clusters: %w", err))
+	}
+	if clusters.Items != nil && len(*clusters.Items) == 0 {
+		return diag.FromErr(fmt.Errorf("no clusters match the specified filtering criteria"))
```
Comment (cristiGuranIonos) [inline ionoscloud/data_source_k8s_clusters.go:72]: can we add `filters` to the err message?

File: ionoscloud/resource_k8s_cluster_test.go [hunk 4]
```diff
@@ -144,6 +142,56 @@ func TestAccK8sClusterPrivate(t *testing.T) {
...
+func TestAccK8sClusters(t *testing.T) {
+	var k8sCluster ionoscloud.KubernetesCluster
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      testAccCheckK8sClusterDestroyCheck,
+		Steps: []resource.TestStep{
```
Comment (cristiGuranIonos) [inline ionoscloud/resource_k8s_cluster_test.go:154]: maybe add an error case, where the filters don't match anything

Comment (cbeti-ionos) [reply ionoscloud/resource_k8s_cluster_test.go:155]: added a no match case ([f6d84ca](https://github.com/ionos-cloud/terraform-provider-ionoscloud/pull/519/commits/f6d84ca77aa5f5db67d18e25d54353ec8a480c4a))

File: ionoscloud/data_source_k8s_clusters.go (hunk 3 above)
Comment (cbeti-ionos) [reply ionoscloud/data_source_k8s_clusters.go]: the filters that were used are now displayed in the error message ([f6d84ca](https://github.com/ionos-cloud/terraform-provider-ionoscloud/pull/519/commits/f6d84ca77aa5f5db67d18e25d54353ec8a480c4a))

**Review by cristiGuranIonos — APPROVED** [review]

**Pull request merged**